src/StreamJsonRpc/MessageHandlerBase.cs

```diff
@@ -35,6 +35,16 @@ public abstract class MessageHandlerBase : IJsonRpcMessageHandler, IDisposableOb
         /// </summary>
         private readonly AsyncSemaphore sendingSemaphore = new AsyncSemaphore(1);
 
+        /// <summary>
+        /// The sync object.
```

Comment:
Small nit: could we add something about this locking/protecting specifically the state of the MessageHandler?

Reply:
Good idea.